src/resolve/resolved-bus.c

```diff
@@ -1855,162 +1855,173 @@ static const sd_bus_vtable resolve_vtable[] = {
-                                 bus_method_set_link_dnssec,
-                                 SD_BUS_VTABLE_UNPRIVILEGED),
-        SD_BUS_METHOD_WITH_NAMES("SetLinkDNSSECNegativeTrustAnchors",
-                                 "ias",
-                                 SD_BUS_PARAM(ifindex)
-                                 SD_BUS_PARAM(names),
-                                 NULL,,
-                                 bus_method_set_link_dnssec_negative_trust_anchors,
-                                 SD_BUS_VTABLE_UNPRIVILEGED),
-        SD_BUS_METHOD_WITH_NAMES("RevertLink",
-                                 "i",
-                                 SD_BUS_PARAM(ifindex),
-                                 NULL,,
-                                 bus_method_revert_link,
-                                 SD_BUS_VTABLE_UNPRIVILEGED),
-        SD_BUS_METHOD_WITH_NAMES("RegisterService",
-                                 "sssqqqaa{say}",
-                                 SD_BUS_PARAM(name)
-                                 SD_BUS_PARAM(name_template)
-                                 SD_BUS_PARAM(type)
-                                 SD_BUS_PARAM(service_port)
-                                 SD_BUS_PARAM(service_priority)
-                                 SD_BUS_PARAM(serwise_weight)
-                                 SD_BUS_PARAM(txt_datas),
-                                 "o",
-                                 SD_BUS_PARAM(service_path),
-                                 bus_method_register_service,
-                                 SD_BUS_VTABLE_UNPRIVILEGED),
-        SD_BUS_METHOD_WITH_NAMES("UnregisterService",
-                                 "o",
-                                 SD_BUS_PARAM(service_path),
-                                 NULL,,
-                                 bus_method_unregister_service,
-                                 SD_BUS_VTABLE_UNPRIVILEGED),
-
-        SD_BUS_METHOD("ResetStatistics",
-                      NULL,
-                      NULL,
-                      bus_method_reset_statistics,
-                      SD_BUS_VTABLE_UNPRIVILEGED),
-        SD_BUS_METHOD("FlushCaches",
-                      NULL,
-                      NULL,
-                      bus_method_flush_caches,
-                      SD_BUS_VTABLE_UNPRIVILEGED),
-        SD_BUS_METHOD("ResetServerFeatures",
-                      NULL,
-                      NULL,
-                      bus_method_reset_server_features,
-                      SD_BUS_VTABLE_UNPRIVILEGED),
+        SD_BUS_METHOD_WITH_ARGUMENTS("ResolveHostname",
+                                     SD_BUS_ARGUMENTS(
+                                             "i", "ifindex",
+                                             "s", "name",
+                                             "i", "family",
+                                             "t", "flags"
+                                     ),
+                                     SD_BUS_RESULT(
```

Comment:
Not that it matters much, but with the syntax highlighting of my editor, I find the definition easier to parse visually if the parameters names are not strings but identifiers:
```
                                     SD_BUS_ARGUMENTS(
                                             "i", ifindex,
                                             "s", name,
                                             "i", family,
                                             "t", flags
                                     ),
```
This can be realized applying STRINGIFY() before concatenation.

Reply:
That actually looks pretty great! Being able to put smaller definitions on a single line is definitely less of a hassle. It kinda depends on a specific kind of syntax highlighting though which is fine for me but maybe not for everyone. @poettering @keszybz thoughts?

Reply:
I am using Emacs without anything fancy. I think the readability is preserved as long as identifiers ans string are highlighted in a different way. But it is matter of taste, I agree.

Reply:
Any way to only allow identifiers without quotes to be passed as argument names? If I stringify them, I don't get any compile errors when passing a string literal with quotes. Not a big problem, but would be nice if we could enforce the separation somehow.

Example: `SD_BUS_ARGUMENTS("i", "ifindex")` still compiles with your approach.